### PR TITLE
Improve readme and adjust changelog

### PR DESCRIPTION
**Pull request by kasium**

### CODE REVIEW

**Comment by kasium** [comment]
TODO: Add boolean migration guide

File: CHANGES.rst [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
-  Please consider a migration from VARCHAR to NVARCHAR.
-- **Backward incompatible change:** Columns of SQLAlchemy type Text are now created with the HANA SQL type NCLOB instead of CLOB.
-  Like the HANA SQL type VARCHAR, CLOB was designed for 7-bit ASCII character data. This change also ensures consistency and
-  compatibility with SAP HANA Cloud, where CLOB is just an alias for NCLOB. Please consider a migration from CLOB to NCLOB.
+- Columns of SQLAlchemy type Text are now created with the HANA SQL type NCLOB instead of CLOB.
+  Like the HANA SQL type VARCHAR, CLOB was designed for 7-bit ASCII character data.
+  This change also ensures consistency and compatibility with SAP HANA Cloud, where CLOB is just an
+  alias for NCLOB
+- Removed ``pyhdb`` support because  ``pyhdb`` is out of maintenance and the GitHub repository was
+  archived. Please migrate to ``hdbcli``
```
Comment (jarus) [inline CHANGES.rst:23]: Please migrate to ``hdbcli`` as it also supports connections towards HANA Cloud databases.

File: CHANGES.rst [hunk 2]
```diff
@@ -1,23 +1,53 @@
...
+- Removed support for python versions below version 3.8
```
Comment (jarus) [inline CHANGES.rst:24]: Python

File: CHANGES.rst [hunk 3]
```diff
@@ -1,23 +1,53 @@
...
+- Removed support for SQLAlchemy below version 1.4
+- Removed support for hdbcli below version 2.10
+- Removed the hidden and outdated feature ``auto_convert_lobs``
+
+Features
+~~~~~~~~
+- Official support for SQLAlchemy 1.4 and 2.0
+- Official support for Python 3.11 and 3.12
+- Added CI with linters and testing utilizing the SQLAchemy and alembic test suite
```
Comment (jarus) [inline CHANGES.rst:33]: Alembic

File: CHANGES.rst [hunk 4]
```diff
@@ -1,23 +1,53 @@
...
+- Support the SAP HANA datatype ``SMALLDECIMAL``
+- Support native booleans (this is the new default)
+- sqlalchemy-hana is fully typed and exports its types
```
Comment (jarus) [inline CHANGES.rst:36]: The ``sqlalchemy_hana`` package is fully typed and exports its types

File: CHANGES.rst (hunk 3 above)
Comment (jarus) [inline CHANGES.rst:33]: As this is mainly a SQLAlchemy-HANA developer feature, I think we should move it a bit lower. It also spoils the Alembic support. ;)

File: CHANGES.rst [hunk 5]
```diff
@@ -1,23 +1,53 @@
...
+- The alembic dialect left the preview stage and is now included by default
```
Comment (jarus) [inline CHANGES.rst:37]: How about pushing the Alembic Support a bit? Which versions of Alembic do we support?

File: CHANGES.rst [hunk 6]
```diff
@@ -1,23 +1,53 @@
...
+- Specified the caching support explicitly (the value might change in the future)
```
Comment (jarus) [inline CHANGES.rst:38]: Maybe highlight that we are talking about SQLAlchemy statement caching. How about mentioning the current value/setting?

File: CHANGES.rst [hunk 7]
```diff
@@ -1,23 +1,53 @@
...
+- Support ``regexp_match`` and ``regexp_replace``
+- Allow usage of "is_distinct_from" operator through a SAP HANA compatible expression
```
Comment (jarus) [inline CHANGES.rst:40]: How about replacing the double quotes with ticks for consistent formatting? ``is_distinct_from``

File: CHANGES.rst [hunk 8]
```diff
@@ -1,23 +1,53 @@
...
+- Prefer dialect types in ``get_columns``
```
Comment (jarus) [inline CHANGES.rst:41]: Maybe reference to the SQLAlchemy documentation?

File: CHANGES.rst [hunk 9]
```diff
@@ -1,23 +1,53 @@
...
+- Allow usage of additional options (e.g. ``nowwait``) in ``with_for_update`` when using
```
Comment (jarus) [inline CHANGES.rst:42]: nowwait? 😉

Comment (adlaube) [inline CHANGES.rst]: nowait

Comment (kasium) [reply CHANGES.rst]: Already fixed

**Review by jarus — APPROVED** [review]

**Pull request merged**